app/src/main/java/com/example/mensfashion/core/base/BaseFragment.kt

```diff
@@ -6,18 +6,28 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.ViewDataBinding
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import androidx.navigation.NavController
+import androidx.navigation.fragment.findNavController
 import com.example.mensfashion.App
 
-abstract class BaseFragment<bindingObj : ViewDataBinding> : Fragment() {
+abstract class BaseFragment<bindingObj : ViewDataBinding,vm:ViewModel> : Fragment() {
     // set view binding in the child class
     abstract fun setViewBinding(): bindingObj
+    // set viewModel in child
+    abstract fun initializeViewModel():vm
+    lateinit var navController: NavController
 
     private var _binding: bindingObj? = null
     val binding: bindingObj get() = _binding!!
     val pref = App.pref
 
+    lateinit var viewModel: vm
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         _binding = setViewBinding()
+        viewModel = initializeViewModel()
+        navController=findNavController()
+        // can't understander what next 3 line do
```

Comment:
please find this anas
for it.executePendingBindings()  =>  https://stackoverflow.com/a/53043911
for => it.lifecycleOwner = viewLifecycleOwner
https://stackoverflow.com/questions/62591052/should-i-add-binding-lifecycleowner-this-when-i-use-viewmodel